qemu/tests/nic_teaming.py

```diff
@@ -0,0 +1,159 @@
+import time
+import re
+import random
+import logging
+from virttest import utils_test, utils_net
+from autotest.client.shared import error, utils
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    Test failover by team driver
+
+    1) Boot a vm with 4 nics.
+    2) inside guest, configure the team driver.
+    3) inside guest, ping host
+    4) inside guest, repeated down the slaves one by one.
+    5) check ping_result.
+
+    :param test: Kvm test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = int(params.get("login_timeout", 1200))
+    session_serial = vm.wait_for_serial_login(timeout=timeout)
+    ifnames = [utils_net.get_linux_ifname(session_serial,
+                                          vm.get_mac_address(vlan))
+               for vlan, nic in enumerate(vm.virtnet)]
+    nm_stop_cmd = "pidof NetworkManager && service NetworkManager stop; true"
+    session_serial.cmd_output_safe(nm_stop_cmd)
+    team_if = params.get("team_if")
+    # initial
+
+    def team_port_add():
+        """Team0 add ports and return the ip link result for debuging"""
+        for port in ifnames:
+            session_serial.cmd_output_safe("ip addr flush %s" % port)
```

Comment:
move these configuration command to cfg, so that if other os support teamd with different cmd, we just need to update cfg  can make test work.